utils.zsh/functions/check_cross_windows_arm64

```diff
@@ -0,0 +1,57 @@
+autoload -Uz log_info log_status log_error log_debug log_warning
+
+# log_debug 'Checking for apt-get...'
+# if (( ! ${+commands[apt-get]} )) {
+#   log_error 'No apt-get command found. Please install apt'
+#   return 2
+# } else {
+#   log_debug "Apt-get located at ${commands[apt-get]}"
+# }
+
+# local -a _quiet=('' '--quiet')
+
+# if [[ ! -f "`pwd`/llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64/bin/aarch64-w64-mingw32-clang" && 
+#       ! -f "`pwd`/llvm-mingw-20220906-ucrt-macos-universal/bin/aarch64-w64-mingw32-clang"]] then
+#   case ${host_os} {
+#     linux)
+#       log_warning 'Downloading llvm-mingw toolchain'
+#       wget https://github.com/mstorsjo/llvm-mingw/releases/download/20220906/llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64.tar.xz
+#       log_warning 'Expanding llvm-mingw toolchain'
+#       tar -xf llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64.tar.xz
+#       log_warning 'Add llvm-mingw toolchain to PATH'
+#       export PATH="`pwd`/llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64/bin:$PATH"
+#       ;;
+#     macos)
+#       log_warning 'Downloading llvm-mingw toolchain'
+#       wget https://github.com/mstorsjo/llvm-mingw/releases/download/20220906/llvm-mingw-20220906-ucrt-macos-universal.tar.xz
+#       log_warning 'Expanding llvm-mingw toolchain'
+#       tar -xf llvm-mingw-20220906-ucrt-macos-universal
+#       log_warning 'Add llvm-mingw toolchain to PATH'
+#       export PATH="`pwd`/llvm-mingw-20220906-ucrt-macos-universal/bin:$PATH"
+#       ;;
+#   }
+# fi
+
+
+case ${host_os} {
+  linux)
+    if [[ ! -f "`pwd`/llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64/bin/aarch64-w64-mingw32-clang" ]]; then
+      log_warning 'Downloading llvm-mingw toolchain'
+      wget https://github.com/mstorsjo/llvm-mingw/releases/download/20220906/llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64.tar.xz
+      log_warning 'Expanding llvm-mingw toolchain'
+      tar -xf llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64.tar.xz
+    fi
+    log_warning 'Add llvm-mingw toolchain to PATH'
+    export PATH="`pwd`/llvm-mingw-20220906-ucrt-ubuntu-18.04-x86_64/bin:$PATH"
+  ;;
+  macos)
+    if [[ ! -f "`pwd`/llvm-mingw-20220906-ucrt-macos-universal/bin/aarch64-w64-mingw32-clang" ]]; then
+      log_warning 'Downloading llvm-mingw toolchain'
+      wget https://github.com/mstorsjo/llvm-mingw/releases/download/20220906/llvm-mingw-20220906-ucrt-macos-universal.tar.xz
+      log_warning 'Expanding llvm-mingw toolchain'
+      tar -xf llvm-mingw-20220906-ucrt-macos-universal
```

Comment:
Do not extract into project directory - if additional 3rd party tools are installed, they should be put in the correct places on the system, in this case either `/usr/bin/` or `/usr/local/bin`, which also would not make `PATH` adjustments necessary.

Reply:
According to https://github.com/mstorsjo/llvm-mingw/issues/250, it should be used in this way.

This toolchain is unpacked, if we just extract the toolchain to anywhere that requires `sudo`, that would be a mess. Keep in mind that CI is not the only end user of this script. You don't want to ruin other people's machine.

Reply:
We already install mingw64 and other build-time dependencies in the system per Linux standards, llvm-mingw should be installed in the same way. That would also remove the requirement to add an additional directory to the `gitignore` file (which is already the first sign that it's not a good practise).

We also don't support cross compiling on macOS (runners are too slow and not available enough for that to make sense), so you can remove that entirely.

More things:
* Use `curl` (which is the canonical tool we use to fetch things and available out of the box on macOS) instead of `wget`.
* Use the `extract` function from the zsh-framework to extract the archive

If you want/need temporary directories, use the `mktemp` function.